File: slidepazzle/cpp/check.cpp
#include<iostream>
#include<vector>
#include"Pazzle.cpp"
char cal[5000][400];
int main()
{
    vector<Pazzle> ps = readfromfile();
    FILE *fp = fopen("submit.txt", "r");
    int nInvalid = 0;
    for (int i = 0; i < 5000; i++)
    {
        if (!strcmp(cal[i], ""))
            continue;
        for (int j = 0; cal[i][j] != '\0'; j++)
            ps[i].operate(VAL[cal[i][j]]);
        if (!ps[i].isComplete())
        {
            printf("error: Problem No=%d\n", i+1);
            cout << ps[i].toString() << endl;
            nInvalid++;
        }
    }
    printf("%d errors.\n", nInvalid);
    return 0 ;
}

File: slidepazzle/cpp/check.py
import sys
import Pazzle
ifilename ='submit.txt'
if len(sys.argv) > 1:
    ifilename = sys.argv[1]

limits,n,ps = Pazzle.readfromfile()
a=0
for i,seq in enumerate(open(ifilename).readlines()):
    seq = seq.strip()
    if seq == '':
        continue
    for ch in seq:
        ps[i].operate(Pazzle.OPS[ch])
    if not ps[i].isComplete():
        print 'error: Problem No=%d %s' % (i+1, ps[i].operations())
        print ps[i]
        a += 1
print '%d errors in %s.' % (a, ifilename)
